lib/models/order_book.js

```diff
@@ -0,0 +1,75 @@
+'use strict'
+
+const { EventEmitter } = require('events')
+
+/**
+ * High level OB model to automatically integrate WS updates & maintain sort
+ */
+class OrderBook extends EventEmitter {
+  /**
+   * Initializes the order book with an existing snapshot (array form)
+   * @param {Array[]} snapshot
+   */
+  constructor (snapshot) {
+    super()
+
+    this.bids = []
+    this.asks = []
+
+    for (let i = 0; i < snapshot.length; i++) {
+      if (snapshot[i][2] < 0) {
+        this.asks.push(snapshot[i])
+      } else {
+        this.bids.push(snapshot[i])
+      }
+    }
+  }
+
+  /**
+   * Integrate an update packet; emits an 'update' event on success
+   * @param {Array} entry
+   */
+  updateWith (entry) {
+    const [price, count, amount] = entry
+    const side = amount < 0 ? this.asks : this.bids
+
+    if (count === 0) {
+      for (let i = 0; i < side.length; i++) {
+        if (side[i][0] === price) {
+          side.splice(i, 1)
+          return
+        }
+      }
+
+      throw new Error(`received rm entry for unknown price lvl: ${price}`)
```

Comment:
are we handling this somewhere? if not  it will crash people's bots

Reply:
The data might be inconsistent due to dropped packets/network troubles; this would kill the bot there to prevent any issues. I switched it to a debug statement for now 🙈 What do you think about maybe clearing the OB if inconsistency is detected?